### PR TITLE
Support Xcode Bots

### PR DESCRIPTION
**Pull request by q231950**
This adds the .xcodeproj to the repository so that Xcode Bots can run and make changes to its schemes. The pull request also separates the Tests for the Core from the Test for the example usage.

The Bots need to have a Bootstrap action in order to prepare the project file. When running locally, it is also needed to bootstrap the project `swift package generate-xcodeproj`.

screen shot 2018-01-20 at 11 57 45

### CODE REVIEW

**Comment by q231950** [comment]
Some adaptions need to be made to a test that tests the Package when run from the command line. 

The Xcode Bot seems to first run
`xcodebuild build-for-testing -scheme 'Commands-Package' -configuration 'Debug'`
then 
`xcodebuild test-without-building -scheme 'Commands-Package' -configuration 'Debug'`

This leads to an infinitely running test case `testExecutesScript` as it runs into a [readline()](https://github.com/q231950/commands/blob/9076cd80a18bcc4cf687052428bb44afc1ef9875/Sources/CommandsCore/Commands.swift#L25). When running tests from Xcode the test runner seems to quit the test.

**Comment by q231950** [comment]
The main issue for testing the Example usage was the fact that the [readline](https://github.com/q231950/commands/pull/18/files#diff-4e4007449fb4cde4a8133de0fc7da12cR26) would wait forever when running the test with `xcodebuild`. It is now solved by allowing the process to [terminate](https://github.com/q231950/commands/pull/18/files#diff-4e4007449fb4cde4a8133de0fc7da12cR19) the run.

File: README.md [hunk 1]
```diff
@@ -6,6 +6,10 @@
 
 `swift package generate-xcodeproj`
 
+Open the project and run the Commands target. The target is configured to run an example with Ruby IRB.
```
Comment (BasThomas) [inline README.md]: Let's either:

- spell out IRB fully
- add a URL to some more information on it.

**Review by BasThomas — APPROVED** [review]
Looks good to me

File: README.md (hunk 1 above)
Comment (q231950) [reply README.md]: I don't know what IRB stands for.. something with interactive probably.. and Ruby... I will fix it.

Comment (BasThomas) [reply README.md]: > The abbreviation irb comes from the fact that the filename extension for Ruby is ".rb", although interactive Ruby files do not have an extension of ".irb".

— https://en.m.wikipedia.org/wiki/Interactive_Ruby_Shell

**Pull request merged**